backend/service/authn/sql_test.go

```diff
@@ -0,0 +1,28 @@
+package authn
+
+import (
+	"context"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateOrUpdateUser(t *testing.T) {
+	db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+	assert.NoError(t, err)
+	defer db.Close()
+
+	id := "foo@example.com"
+	refreshToken := []byte("abdcdefgzzz")
+
+	r := &repository{db: db}
+	mock.ExpectQuery(createOrUpdateUser).
```

Comment:
I believe your missing an assert for mock at the end of your test, `mock.ExpectationsWereMet()`.

For example: https://github.com/DATA-DOG/go-sqlmock/blob/master/examples/basic/basic_test.go#L28-L31